lib/update-builder.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const { QueryConditionsBuilder } = require('./query-conditions-builder');
+const { mapJoinIterable } = require('./utils');
+const { SelectBuilder } = require('./select-builder');
+const { makeParamValue } = require('./query-builder');
+
+class UpdateBuilder extends QueryConditionsBuilder {
+  constructor(params, options) {
+    super(params, options);
+    this.operations = {
+      table: null,
+      from: [],
+      set: [],
+    };
+  }
+
+  table(tableName, alias) {
+    const table = this.escapeIdentifier(tableName);
+    this.operations.table = { table, alias: alias && this.escapeKey(alias) };
+    return this;
+  }
+
+  set(key, value) {
+    this.operations.set.push({
+      column: this.escapeKey(key),
+      value: this._whereValueMapper(value),
+    });
+  }
+
+  from(tableName, alias) {
+    const table = this.escapeIdentifier(tableName);
+    this.operations.from.push({ table, alias: alias && this.escapeKey(alias) });
+    return this;
+  }
+
+  _whereValueMapper(subquery) {
+    return typeof subquery === 'function'
+      ? subquery(new SelectBuilder(this.params, this.options))
+      : subquery;
+  }
+
+  _processFrom(from) {
+    return mapJoinIterable(
+      from,
+      f => (!f.alias ? String(f.table) : `${f.table} AS ${f.alias}`),
+      ', '
+    );
+  }
+
+  _processSet(set) {
+    return mapJoinIterable(
+      set,
+      s => `${s.column} = ${makeParamValue(s.value, this.params)}`,
+      ', '
+    );
+  }
+
+  build() {
+    let query = 'UPDATE';
```

Comment:
Maybe: query = []; .push, push, .join ?

Reply:
I prefer to avoid array-join unnecessarily.